### PR TITLE
Npm test

### LINKED ISSUE #1002: [npm-installer] add lints/tests for the raw package
The npm template is now a perfectly cromulent and valid npm package which can have `npm run fmt:check` and such run on it.

We should:

* tell dependabot about it
* add a ci workflow that does `npm ci && npm run fmt:check` on it

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #1002

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,6 +20,17 @@ env:
...
+  npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v4
+        with:
+            node-version: 20
+      - run: npm ci
+      - run: npm run fmt:check
+
```
Comment (Gankra) [inline .github/workflows/ci.yml:33]: presumably this needs to cd into the relevant dir, but otherwise rad

**Review by Gankra — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mistydemeo) [reply .github/workflows/ci.yml:33]: The subdir is `cargo-dist/templates/installer/npm` for reference

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,6 +20,17 @@ env:
...
 jobs:
+  # Test and fmt the npm pkg
+  npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v4
+        with:
+            node-version: 20
+      - run: npm ci
```
Comment (mistydemeo) [inline .github/workflows/ci.yml:31]: Try this:
```suggestion
      - run: npm ci
        working-directory: cargo-dist/templates/installer/npm
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -20,6 +20,17 @@ env:
...
+      - run: npm run fmt:check
```
Comment (mistydemeo) [inline .github/workflows/ci.yml:32]:
```suggestion
      - run: npm run fmt:check
        working-directory: cargo-dist/templates/installer/npm
```

**Review by mistydemeo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -20,6 +20,18 @@ env:
...
 
 jobs:
+  # Test and fmt the npm pkg
+  npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v4
+      - name: npm fmt
+        with:
```
Comment (mistydemeo) [inline .github/workflows/ci.yml:30]: I think this belongs in the step above

Comment (ashleygwilliams) [reply .github/workflows/ci.yml:29]: hrm? the with tells node what version to use

Comment (mistydemeo) [reply .github/workflows/ci.yml:29]: Sorry, I think at the time I submitted it this was attached to the step below. It's fine now!

**Review by mistydemeo — APPROVED** [review]

**Pull request merged**